projects/ADuCM3029_demo_cn0428_cn0429/src/adi_gas_sensor.h

```diff
@@ -0,0 +1,234 @@
+/**
+******************************************************************************
+*   @file     adi_gas_sensor.h
+*   @brief    Gas sensor class object header file
+*   @version  V0.1
+*   @author   ADI
+*
+*******************************************************************************
+*
+*Copyright 2015-2018(c) Analog Devices, Inc.
+*
+*All rights reserved.
+*
+*Redistribution and use in source and binary forms, with or without modification,
+*are permitted provided that the following conditions are met:
+*    - Redistributions of source code must retain the above copyright
+*      notice, this list of conditions and the following disclaimer.
+*    - Redistributions in binary form must reproduce the above copyright
+*      notice, this list of conditions and the following disclaimer in
+*      the documentation and/or other materials provided with the
+*      distribution.
+*    - Neither the name of Analog Devices, Inc. nor the names of its
+*      contributors may be used to endorse or promote products derived
+*      from this software without specific prior written permission.
+*    - The use of this software may or may not infringe the patent rights
+*      of one or more patent holders.  This license does not release you
+*      from the requirement that you obtain separate licenses from these
+*      patent holders to use this software.
+*    - Use of the software either in source or binary form, must be run
+*      on or directly connected to an Analog Devices Inc. component.
+*
+*THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+*INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT, MERCHANTABILITY AND FITNESS FOR A
+*PARTICULAR PURPOSE ARE DISCLAIMED.
+*
+*IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+*EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, INTELLECTUAL PROPERTY
+*RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR
+*BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+*STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+*THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************
+**/
+
+#ifndef ADI_GAS_SENSOR_H_
+#define ADI_GAS_SENSOR_H_
+
+#include "base_sensor/adi_sensor.h"
+#include "adi_m355_gas_sensor_config.h"
+
+namespace adi_sensor_swpack
+{
+/**
+ * @class Temperature.
+ *
+ * @brief Temperature class abstracts the temperature sesor functionality.
+ *
+ **/
+class Gas_Reading : public Sensor
```

Comment:
I think this class is also used for the water sensor so maybe naming it Gas_Reading is not quite correct.

Reply:
Just curious, is there any particular reason why this should be implemented in C++? I think this reduces portability and adds unnecessary overhead.

Reply:
@mirceacaprioru We originally had separate classes for each sensor type. The only members in the water sensors class were open, close, start, stop, read_write, and init functions, all identical to the gas sensor class. But when you turn on the board, you don't know yet whether the sensor is water or gas, so the gas object always gets initialized first and then we determine based on its response what kind of sensor it actually is. This caused conflicts when you tried to initialize a water sensor object as well. We resolved these errors and reduced the number of files by having the water sensors leverage the gas sensor class. In reality, some member functions in the class are still only applicable to gas sensors.

Reply:
@stefpopa We're using the adi_sensors classes, so we have to use C++ because C doesn't support classes.